fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProvider.java

```diff
@@ -77,6 +79,21 @@
             SYSTEM_CONFIG.getPackageVariableName("lucene_search_timeout_ms"), 600000
     );
 
+    public static final LRUQueryCache LUCENE_QUERY_CACHE;
+    static {
+        // Lucene 5.3.0 disables query caching.  Create cache for use here.
+        // min of 32MB or 5% of the heap size.  Same default as Lucene 5.4.0.
+        final long maxRamBytesUsed = SYSTEM_CONFIG.getLongProperty(
+            SYSTEM_CONFIG.getPackageVariableName("lucene_cache_size_bytes"),
+            Math.min(1L << 25, Runtime.getRuntime().maxMemory() / 20));
+
+        // default max cached queries based on memory used
+        final int maxCachedQueries = SYSTEM_CONFIG.getIntProperty(
+            SYSTEM_CONFIG.getPackageVariableName("lucene_cache_queries"),
+            (int) Math.max(1000L, Math.min(1000000L, maxRamBytesUsed / 1000)));
+        LUCENE_QUERY_CACHE = new LRUQueryCache(maxCachedQueries, maxRamBytesUsed);
+    }
+
```

Comment:
They now appear to use the config if set, and otherwise default, so that aspect has been addressed. But I'm still eager to hear why these are the defaults?